decision_service/include/ds_api_status.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <string>
+
+
+namespace decision_service { namespace error_code {
+    //success code
+    const int success = 0;
+
+    //error code
+    const int invalid_argument = 1;
+    const int background_queue_overflow = 2;
+    const int eventhub_http_generic = 3;
+    const int eventhub_http_bad_status_code = 4;
+}}
+
+namespace decision_service {
+  class api_status
+  {
+    public:
+      api_status();
+
+      int get_error_code() const;
+      const char* get_error_msg() const;
+
+      static void try_update(api_status* status, int new_code, const char* new_msg);
```

Comment:
What is the meaning of these functions?

Reply:
They are helper functions used to check for nullptr api_status.  

---
In reply to: [187117708](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r187117708) [](ancestors = 187117708)

Reply:
Ok, sounds good.

Reply:
(Maybe add a comment.)

Reply:
Yep

---
In reply to: [187340727](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r187340727) [](ancestors = 187340727)